source/dem/particle_wall_dmt_force.cc

```diff
@@ -0,0 +1,507 @@
+
+  // Calculation of model parameters (beta, sn and st). These values
+  // are used to consider non-linear relation of the contact force to
+  // the normal overlap
+  const double radius_times_overlap_sqrt =
+    sqrt(effective_radius * contact_info.normal_overlap);
+  const double model_parameter_sn =
+    2 * this->effective_youngs_modulus[particle_type] *
+    radius_times_overlap_sqrt;
+
+  const double model_parameter_st =
+    8 * this->effective_shear_modulus[particle_type] *
+    radius_times_overlap_sqrt;
+
+  // Calculation of normal and tangential spring and dashpot constants
+  // using particle and wall properties
+  const double normal_spring_constant =
+    1.3333 * this->effective_youngs_modulus[particle_type] *
+    radius_times_overlap_sqrt;
+
+  // Calculation of normal damping and tangential spring and dashpot constants
+  //  using particle and wall properties.
+  //  There is no minus sign here since model_parameter_beta is negative or
+  //  equal to zero.
+  const double normal_damping_constant =
+    1.8257 * this->model_parameter_beta[particle_type] * // 2. * sqrt(5./6.)
+    sqrt(model_parameter_sn * particle_properties[DEM::PropertiesIndex::mass]);
+
+  // Tangential spring constant is set as a negative just like in the other
+  // particle-wall models. This must be taken into account for the square root
+  // in the tangential_damping_calculation.
+  double tangential_spring_constant =
+    -8. * this->effective_shear_modulus[particle_type] *
+      radius_times_overlap_sqrt +
+    DBL_MIN;
+
+  // There is no minus sign here since model_parameter_beta is negative or
+  // equal to zero.
+  const double tangential_damping_constant =
+    normal_damping_constant *
+    sqrt(model_parameter_st / (model_parameter_sn + DBL_MIN));
+
+  // Calculation of the normal force coefficient (F_n_DMT)
+  const double cohesive_term = 2. * M_PI * effective_radius *
+                               this->effective_surface_energy[particle_type];
+  double normal_force_norm =
+    normal_spring_constant * contact_info.normal_overlap +
+    normal_damping_constant * contact_info.normal_relative_velocity;
+
+  // Calculation of tangential force
+  Tensor<1, 3> damping_tangential_force =
+    tangential_damping_constant * contact_info.tangential_relative_velocity;
+  Tensor<1, 3> tangential_force =
+    tangential_spring_constant * contact_info.tangential_overlap +
+    damping_tangential_force;
+  double tangential_force_norm = tangential_force.norm();
+
+  // Coulomb limit need to be modified (Thornton 1991)
+  double coulomb_threshold =
+    this->effective_coefficient_of_friction[particle_type] * normal_force_norm;
+
+  // Check for gross sliding
+  if (tangential_force_norm > coulomb_threshold)
+    {
+      // Gross sliding occurs and the tangential overlap and tangential
+      // force are limited to Coulomb's criterion
+      contact_info.tangential_overlap =
+        (coulomb_threshold *
```

Comment:
Same comment as before. Too much repeated code, imo

Reply:
Same applies for JKR

Reply:
This will go in an other PR